receiver/dcgmreceiver/scraper.go

```diff
@@ -90,44 +95,144 @@ func (s *dcgmScraper) scrape(_ context.Context) (pmetric.Metrics, error) {
-			s.mb.RecordDcgmGpuProfilingPipeUtilizationDataPoint(now, metric.asFloat64(), model, gpuIndex, UUID, metadata.AttributePipeFp64)
-		case "dcgm.gpu.profiling.fp32_utilization":
-			s.mb.RecordDcgmGpuProfilingPipeUtilizationDataPoint(now, metric.asFloat64(), model, gpuIndex, UUID, metadata.AttributePipeFp32)
-		case "dcgm.gpu.profiling.fp16_utilization":
-			s.mb.RecordDcgmGpuProfilingPipeUtilizationDataPoint(now, metric.asFloat64(), model, gpuIndex, UUID, metadata.AttributePipeFp16)
-		case "dcgm.gpu.profiling.dram_utilization":
-			s.mb.RecordDcgmGpuProfilingDramUtilizationDataPoint(now, metric.asFloat64(), model, gpuIndex, UUID)
-		case "dcgm.gpu.profiling.pcie_sent_bytes":
-			/* DCGM already returns these as bytes/sec despite the name */
-			s.mb.RecordDcgmGpuProfilingPcieTrafficRateDataPoint(now, metric.asInt64(), model, gpuIndex, UUID, metadata.AttributeDirectionTx)
-		case "dcgm.gpu.profiling.pcie_received_bytes":
-			s.mb.RecordDcgmGpuProfilingPcieTrafficRateDataPoint(now, metric.asInt64(), model, gpuIndex, UUID, metadata.AttributeDirectionRx)
-		case "dcgm.gpu.profiling.nvlink_sent_bytes":
-			s.mb.RecordDcgmGpuProfilingNvlinkTrafficRateDataPoint(now, metric.asInt64(), model, gpuIndex, UUID, metadata.AttributeDirectionTx)
-		case "dcgm.gpu.profiling.nvlink_received_bytes":
-			s.mb.RecordDcgmGpuProfilingNvlinkTrafficRateDataPoint(now, metric.asInt64(), model, gpuIndex, UUID, metadata.AttributeDirectionRx)
+			s.mb.RecordGpuDcgmUtilizationDataPoint(now, gpuUtil)
+		}
+		if metric, ok := metrics["DCGM_FI_PROF_SM_ACTIVE"]; ok {
+			s.mb.RecordGpuDcgmSmUtilizationDataPoint(now, metric.asFloat64())
+		}
+		if metric, ok := metrics["DCGM_FI_PROF_SM_OCCUPANCY"]; ok {
+			s.mb.RecordGpuDcgmSmOccupancyDataPoint(now, metric.asFloat64())
+		}
+		if metric, ok := metrics["DCGM_FI_PROF_PIPE_TENSOR_ACTIVE"]; ok {
+			s.mb.RecordGpuDcgmPipeUtilizationDataPoint(now, metric.asFloat64(), metadata.AttributeGpuPipeTensor)
+		}
+		if metric, ok := metrics["DCGM_FI_PROF_PIPE_FP64_ACTIVE"]; ok {
+			s.mb.RecordGpuDcgmPipeUtilizationDataPoint(now, metric.asFloat64(), metadata.AttributeGpuPipeFp64)
+		}
+		if metric, ok := metrics["DCGM_FI_PROF_PIPE_FP32_ACTIVE"]; ok {
+			s.mb.RecordGpuDcgmPipeUtilizationDataPoint(now, metric.asFloat64(), metadata.AttributeGpuPipeFp32)
+		}
+		if metric, ok := metrics["DCGM_FI_PROF_PIPE_FP16_ACTIVE"]; ok {
+			s.mb.RecordGpuDcgmPipeUtilizationDataPoint(now, metric.asFloat64(), metadata.AttributeGpuPipeFp16)
+		}
+		if metric, ok := metrics["DCGM_FI_DEV_ENC_UTIL"]; ok {
+			encUtil := float64(metric.asInt64()) / 100.0 /* normalize */
+			s.mb.RecordGpuDcgmCodecEncoderUtilizationDataPoint(now, encUtil)
+		}
+		if metric, ok := metrics["DCGM_FI_DEV_DEC_UTIL"]; ok {
+			decUtil := float64(metric.asInt64()) / 100.0 /* normalize */
+			s.mb.RecordGpuDcgmCodecDecoderUtilizationDataPoint(now, decUtil)
+		}
+		if metric, ok := metrics["DCGM_FI_DEV_FB_FREE"]; ok {
+			bytesFree := 1e6 * metric.asInt64() /* MBy to By */
```

Comment:
None of this is very well documented, but yes, this has the type `DCGM_FT_INT64`, and is thus measured in whole megabytes. The type was determined empirically.
Sadly, we can't use `asFloat64` here, because that would just give us nonsense[^1].

[^1]: https://github.com/NVIDIA/go-dcgm/blob/main/pkg/dcgm/fields.go#L189-L191